src/components/load-safe/steps/SafeOwnersStep.tsx

```diff
@@ -39,24 +39,14 @@ const SafeOwnersStep = ({ params, onSubmit, onBack }: Props): ReactElement => {
     setValue('threshold', safeInfo.threshold)
     setValue(
       'owners',
-      safeInfo.owners.map((owner) => ({ address: owner.value, name: '', resolving: false })),
+      safeInfo.owners.map((owner) => ({ address: owner.value, name: '' })),
     )
   }, [safeInfo, setValue])
 
-  const onFormSubmit = handleSubmit((data: SafeFormData) => {
-    onSubmit({
-      ...data,
-      owners: data.owners.map((owner) => ({
-        name: owner.name || owner.fallbackName,
-        address: owner.address,
-      })),
-    })
-  })
-
   return (
     <Paper>
       <FormProvider {...formMethods}>
-        <form onSubmit={onFormSubmit}>
+        <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
Adding a name for an owner, going to the next step and going back doesn't preserve the name anymore. It still seems to work in the create safe flow.

Reply:
Do you have owner saved in the address book?

Reply:
Should be fixed now.